src/fns.js

```diff
@@ -80,27 +80,21 @@ function matchRoute(routes, path) {
  * Given a list of matching routes, return a promise which resolves
  * with the first route in the list that has indicated it should handle
  * based on its shouldHandle key.
- * @param routes
+ * @param {Object[]} routes
  * @returns {Promise}
  */
 function filterMatches(routes) {
-  return new Promise((res, rej) => {
-    if (routes.length === 0) {
-      rej('No routes match');
-    } else if (routes.length === 1) {
-      res(routes[0]);
-    } else {
-      utils.PromiseOrderedFirst(
-        routes.map(r => {
-          if (r.route.shouldHandle) {
-            return r.route.shouldHandle();
-          }
-          // The absence of route.shouldHandle is considered a match.
-          return Promise.resolve();
-        })
-      ).then(({index}) => res(routes[index]));
-    }
-  });
+  return utils
```

Comment:
will this rej if no routes match?

Reply:
Yup. Theres a test for that too.

`PromiseOrderedFirst` will iterate through the (empty) list of promises and since it wont find any pending or resolved ones, it will reject on first attempt.